packages/segmented-control/src/SegmentedControlOption.tsx

```diff
@@ -0,0 +1,238 @@
+    --base-background-color: transparent;
+    --base-shadow-color: transparent;
+    --hover-text-color: ${uiColors.gray.light2};
+    --hover-background-color: ${uiColors.gray.dark2};
+    --active-text-color: ${uiColors.white};
+    --disabled-text-color: ${uiColors.gray.dark1};
+  `}
+
+  position: relative;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  padding: var(--inline-padding) 12px;
+  border-radius: 4px;
+  text-align: center;
+  font-size: var(--font-size);
+  line-height: var(--line-height);
+  text-transform: var(--text-transform, none);
+  font-weight: var(--font-weight);
+  color: var(--base-text-color); // color
+  background-color: var(--base-background-color); // color
+  box-shadow: 0px 1px 2px var(--base-shadow-color); // color
+  cursor: pointer;
+  transition: all 100ms ease-in-out;
+  z-index: 1;
+
+  &:hover {
+    color: var(--hover-text-color);
+
+    &:after {
+      background-color: var(--hover-background-color);
+    }
+
+    &${checkedSelector}, &[data-disabled='true'] {
+      border-color: transparent;
+      background-color: transparent;
+
+      &:after {
+        background-color: transparent;
+      }
+    }
+  }
+
+  &${checkedSelector} {
+    color: var(--active-text-color); // color
+  }
+
+  &[data-disabled='true'] {
+    color: var(--disabled-text-color); // color
+    cursor: not-allowed;
+  }
+
+  // Hover indicator
+  &:after {
+    content: '';
+    position: absolute;
+    height: calc(100% - 2px);
+    width: 100%;
+    top: 1px;
+    left: 0;
+    background-color: transparent;
+    z-index: -1;
+    border-radius: inherit;
+    transition: all 100ms ease-in-out;
+  }
+
+  /* 
+   * Adds the divider line to unselected segments 
+   */
+  &:before {
+    content: '';
+    position: absolute;
+    height: var(--divider-height);
+    width: 1px;
+    left: calc(0px - (var(--segment-gap, 1px) + 1px) / 2);
+    background-color: transparent;
+    transition: all 100ms ease-in-out;
+  }
+
+  &${uncheckedSelector} {
+    &:not(:first-child) {
+      &:not(${checkedSelector} + ${uncheckedSelector}) {
+        // no divider to the left of the checked segment
+        &:before {
+          background-color: ${uiColors.gray.light1}; // color
+        }
+      }
+    }
+  }
+`;
+
+const radioInputStyle = css`
+  height: 0;
+  width: 0;
+  margin: 0;
+  opacity: 0;
+  pointer-events: none;
+`;
+
+const labelStyle = css`
+  display: inline-flex;
+  align-items: center;
+  gap: calc(var(--font-size) / 2);
+  cursor: inherit;
+  z-index: 3;
+`;
+
+/**
+ * Types
+ */
+
+// Internal “Private” Props
+// _name - the name of the segmented control. Used to group the HTML radio inputs
+// _checked - whether the option is checked. Defined by the parent
+// _onChange - the onChange handler passed in from the SegmentedControl
+export interface SegmentedControlOptionProps {
+  value: string;
+  children: React.ReactNode;
+  disabled?: boolean;
+  as?: any;
```

Comment:
Thinking more on this, I'm not sure I see a whole lot of value compared to the complexity of supporting an `as` prop on the option.

The main use case I see for the `as` prop is to support routing components like (Next.js `Link`), but since (a) this can be done programmatically via `router.push`, and (b) since we don't want to encourage using the segmented control for navigation, I don't think we should support this functionality

Reply:
The value is primarily in accessibility and supporting edge cases we didn't anticipate. `router.push` requires a ton of manual work to make it translate well to screen readers, support keybindings like command+click to open in a new tab, etc. We have an entire component built to make the typing and implementation of `as` easier, and I feel strongly we should support this functionality.